internal/pkg/aws/aws.go

```diff
@@ -14,15 +14,36 @@ import (
 type api interface {
```

Comment:
I think it's a little confusing to call this package aws, especially because you are importing aws-sdk-go/aws . Can this package be named something else. maybe 'api' ? Also, can we not use the aws-sdk-go-v2 packages here? I suppose that is outside the scope of this specific pull request.